### PR TITLE
Map query -> scouter

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

**Review by Yotam7 — APPROVED** [review]

File: vite-project/src/scouter/ScouterQuery.tsx [hunk 1]
```diff
@@ -16,23 +16,23 @@ const ScouterQuery: React.FC<ScouterQueryProps> = ({
   list,
 }) => {
   function renderInput() {
-    const initialValue = localStorage.getItem(name) || "";
-
+    const queryName = "Queries/" + name;
```
Comment (YoniKiriaty) [inline vite-project/src/scouter/ScouterQuery.tsx:19]: `Queries/${name}`

File: vite-project/src/scouter/ScouterApp.tsx [hunk 2]
```diff
@@ -15,10 +16,24 @@ function ScouterApp() {
       formValues[key] = value.toString();
     }
     setFormData(formValues);
+    clearQueryStorage();
+  }
+
+  function clearQueryStorage() {
```
Comment (YoniKiriaty) [inline vite-project/src/scouter/ScouterApp.tsx:22]: Object.keys(localStorage)
 .filter(x =>
    x.startsWith('TM_'))
 .forEach(x => 
    localStorage.removeItem(x))

File: vite-project/src/scouter/querytypes/MapQuery.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+interface MapQueryProps {
+  name: string;
+  width: number;
+  height: number;
+  imagePath: string;
+  primaryButtons: Record<string, string>;
+  secondaryButtons?: Record<string, string[]>;
+}
+interface Point {
```
Comment (YoniKiriaty) [inline vite-project/src/scouter/querytypes/MapQuery.tsx:11]: move to common file

File: vite-project/src/scouter/querytypes/MapQuery.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  x: number;
+  y: number;
+  data: Record<string, string>;
```
Comment (YoniKiriaty) [inline vite-project/src/scouter/querytypes/MapQuery.tsx:14]: data?

File: vite-project/src/scouter/querytypes/MapQuery.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  name,
+  width,
+  height,
+  imagePath,
+  primaryButtons,
+  secondaryButtons,
+}) => {
+  const localStorageKey = "Queries/" + name + "/Points";
+  const [points, setPoints] = useState<Point[]>(
+    JSON.parse(localStorage.getItem(localStorageKey) || "[]")
```
Comment (YoniKiriaty) [inline vite-project/src/scouter/querytypes/MapQuery.tsx:29]: try to make this better

**Review by YoniKiriaty — COMMENTED** [review]
John Review

**Pull request merged**